src/ashpy/losses/gan.py

```diff
@@ -114,7 +114,9 @@ def get_discriminator_inputs(
 class GeneratorAdversarialLoss(GANExecutor):
     r"""Base class for the adversarial loss of the generator."""
 
-    def __init__(self, loss_fn: tf.keras.losses.Loss = None) -> None:
+    def __init__(
+        self, loss_fn: tf.keras.losses.Loss = None, name="GeneratorAdversarialLoss"
```

Comment:
Please add Type Annotations to both signature and docs

Reply:
Also very important: must this `name` argument be unique? Does it behave like the `name` argument for `Metric` and `Callback`?

Reply:
Yes, the loss name should be unique. We actually log the loss as ashpy/loss_name.

Reply:
Where can we specify the fact that the name should be unique?

Reply:
In the documentation + we must keep track of every loss passed to a trainer, store the names and check for duplicates. If there are duplicates, raise an exception

Reply:
Add check in trainers, need to add documentation.